src/react-components/ui-root.js

```diff
@@ -482,6 +498,132 @@ class UIRoot extends Component {
     this.setState({ exited: true });
   };
 
+  emojiChange = reason => {
```

Comment:
As mentioned previously, this code could be simplified quite a bit if emojiChange just set a single state variable and emitted a single event:
```javascript
emojiChange = emojiType => {
  const newEmoji = this.state.currentEmoji === emojiType ? null : emojiType;
  this.setState({currentEmoji: newEmoji});
  this.props.scene.emit("change_emoji", {emojiType: newEmoji});
};
```